src/components/hooks/useFindProductInDelivered.ts

```diff
@@ -0,0 +1,58 @@
+import { useQuery } from "@tanstack/react-query";
+import { fetchDelivered } from "../api/api.ts";
+
+type HookResponse =
+  | undefined
+  | {
+      daysDifference: number;
+      parsedOrderDate: Date;
+      matchingItem: {
+        id: string;
+        amount: number;
+      };
+    };
+
+export const useFindProductInDelivered = (
+  productIds: string[],
+): HookResponse => {
+  const { data, isError } = useQuery(["data"], () => fetchDelivered());
+
+  if (isError) {
+    throw new Error("Error fetching data.");
```

Comment:
Nelibi se mi ergonomie hooku, co hazi chybu, meli bychom vymyset nejaky zpusob jak to lip handlovat.